src/test/java/api/loans/CheckOutByBarcodeScheduledNoticesTests.java

```diff
@@ -0,0 +1,232 @@
+import api.support.builders.NoticePolicyBuilder;
+import io.vertx.core.json.JsonObject;
+
+public class CheckOutByBarcodeScheduledNoticesTests extends APITests {
+
+  private static final String BEFORE_TIMING = "Before";
+  private static final String UPON_AT_TIMING = "Upon At";
+  private static final String AFTER_TIMING = "After";
+
+  @Test
+  public void allDueDateNoticesShouldBeScheduledOnCheckoutWhenPolicyDefinesDueDateNoticeConfiguration()
+    throws MalformedURLException,
+    InterruptedException,
+    TimeoutException,
+    ExecutionException {
+
+    UUID beforeTemplateId = UUID.randomUUID();
+    Period beforePeriod = Period.days(2);
+    Period beforeRecurringPeriod = Period.hours(6);
+
+    UUID uponAtTemplateId = UUID.randomUUID();
+
+    UUID afterTemplateId = UUID.randomUUID();
+    Period afterPeriod = Period.days(3);
+    Period afterRecurringPeriod = Period.hours(4);
+
+    JsonObject beforeDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(beforeTemplateId)
+      .withDueDateEvent()
+      .withBeforeTiming(beforePeriod)
+      .recurring(beforeRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+    JsonObject uponAtDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(uponAtTemplateId)
+      .withDueDateEvent()
+      .withUponAtTiming()
+      .sendInRealTime(false)
+      .create();
+    JsonObject afterDueDateNoticeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(afterTemplateId)
+      .withDueDateEvent()
+      .withAfterTiming(afterPeriod)
+      .recurring(afterRecurringPeriod)
+      .sendInRealTime(true)
+      .create();
+    NoticePolicyBuilder noticePolicy = new NoticePolicyBuilder()
+      .withName("Policy with due date notices")
+      .withLoanNotices(Arrays.asList(
+        beforeDueDateNoticeConfiguration,
+        uponAtDueDateNoticeConfiguration,
+        afterDueDateNoticeConfiguration));
+    useLoanPolicyAsFallback(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicy).getId());
+
+    IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource steve = usersFixture.steve();
+
+    final DateTime loanDate =
+      new DateTime(2018, 3, 18, 11, 43, 54, DateTimeZone.UTC);
+
+    final IndividualResource loan = loansFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(steve)
+        .on(loanDate)
+        .at(UUID.randomUUID()));
+    DateTime dueDate = new DateTime(loan.getJson().getString("dueDate"));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(scheduledNoticeClient::getAll, Matchers.hasSize(3));
+
+    List<JsonObject> scheduledNotices = scheduledNoticeClient.getAll();
+    MatcherAssert.assertThat(scheduledNotices,
```

Comment:
Personally, I would statically import `assertThat` to improve readability

Reply:
Done